tests/end_to_end_tests/run_in_remote_machine/cleanup_test_ec2_objects.py

```diff
@@ -0,0 +1,232 @@
+
+# Age of the prefix entry ofter which it can be cleaned up.
+PREFIX_LIST_ENTRY_REMOVE_THRESHOLD = 60 * 7  # Minutes
+
+# We delete any old automated test nodes which are older than 4 hours
+DELETE_OLD_NODES_TIMEDELTA = datetime.timedelta(hours=4)
+DELETE_OLD_NODES_THRESHOLD_DT = datetime.datetime.utcnow() - DELETE_OLD_NODES_TIMEDELTA
+
+
+def _parse_entry_description(entry: Dict):
+    """
+    Parse json object from the description of the prefix list entry.
+    Conventionally, we store useful information in it.
+    """
+    return json.loads(entry["Description"])
+
+
+def _parse_entry_timestamp(entry: Dict):
+    """
+    Parse creation timestamp of the prefix list entry.
+    """
+    return _parse_entry_description(entry)["time"]
+
+
+def _remove_entries(
+        client,
+        entries: List,
+        prefix_list_id: str
+):
+    """
+    Remove specified entries from prefix list.
+    :param client: boto3 client.
+    :param entries: List of entries to remove.
+    :param prefix_list_id: Prefix list ID.
+    :return:
+    """
+    attempts = 20
+    # Since there may be multiple running ec2 tests, we have to add the retry
+    # logic to overcome the prefix list concurrent access issues.
+    while True:
+        try:
+            version = get_prefix_list_version(
+                client=client,
+                prefix_list_id=prefix_list_id
+            )
+            client.modify_managed_prefix_list(
+                PrefixListId=prefix_list_id,
+                CurrentVersion=version,
+                RemoveEntries=[{"Cidr": e["Cidr"]} for e in entries]
+            )
+            break
+        except botocore.exceptions.ClientError as e:
+            keep_trying = False
+            if "The prefix list has the incorrect version number" in str(e):
+                keep_trying = True
+
+            if "The request cannot be completed while the prefix" in str(e):
+                keep_trying = True
+
+            if attempts == 0 or not keep_trying:
+                raise
+
+            attempts -= 1
+            print(f"Can not modify prefix list, retry. Reason: {str(e)}")
+            time.sleep(random.randint(1, 5))
+
+
+def cleanup_old_prefix_list_entries(
+    client,
+    prefix_list_id: str,
+    workflow_id: str = None
+):
+    """
+    Cleanup ec2 test related prefix lists entries.
+    :param client: boto3 client.
+    :param prefix_list_id: Prefix list ID.
+    :param workflow_id: Workflow id to filter workflow related entries.
+    :return:
+    """
+    resp = boto_client.get_managed_prefix_list_entries(
+        PrefixListId=args.prefix_list_id
+    )
+    entries = resp["Entries"]
+
+    entries_to_remove = {}
+
+    current_time = time.time()
+
+    # Remove old prfix list entries.
+    for entry in entries:
+        timestamp = _parse_entry_timestamp(entry)
+        if timestamp <= current_time - PREFIX_LIST_ENTRY_REMOVE_THRESHOLD:
```

Comment:
May want to ensure timestamp is a float or int in _parse_entry_timestamp to avoid potentially comparing a timestamp as a string to current_time

Reply:
Sure, done.